geoh5py/objects/surveys/direct_current.py

```diff
@@ -118,29 +117,21 @@ def copy(self, parent=None, copy_children: bool = True):
             self, parent, copy_children=copy_children, omit_list=omit_list
         )
         setattr(new_entity, "_ab_cell_id", None)
-
+        if new_entity.ab_cell_id is None and self.ab_cell_id is not None:
```

Comment:
**question:** given the `setattr` call above, is this test `new_entity.ab_cell_id is None` not always true?

Reply:
ho, `ab_cell_id()` is maybe getting the ID from somewhere else. Fair enough.

Reply:
correct yes, the ab_cell_id will go through the property again and try to pull from geoh5. The reason being that if copy_children==True, it will already be available